### DISCUSSION TITLE
[SOLVED] Problem with register

### DISCUSSION THREAD

**Discussion by maitalata** [codeigniter4/shield#400] [Q&A] [answered]
I install shied without any problem and I have made all migrations successfully.  But when i submit the registration form , I get the error 

Class "Config\PasswordRules" not found

**Comment by datamweb** [comment]
@maitalata Hello, welcome to Shield.

First of all I want to know what makes users not use command `php spark shield:setup` to install?
Can you explain why you don't use command `shield:setup`?

**Reply by maitalata** [reply]
I use the  `php spark shield:setup` for the installation

**Reply by datamweb** [reply]
Which version did you install?

**Reply by maitalata** [reply]
I use `composer require codeigniter4/shield:dev-develop`

**Reply by datamweb** [reply]
I can not reproduce.
Can you provide more explanation? (screenshot, or any other action you did during installation or after installation?)

**Comment by maitalata** [comment]
<img width="910" alt="Screenshot 2022-08-22 at 1 38 13 PM" src="https://user-images.githubusercontent.com/16260233/185924905-55c193bb-6aa3-490a-85f1-d26d067c608d.png">
<img width="759" alt="Screenshot 2022-08-22 at 1 38 58 PM" src="https://user-images.githubusercontent.com/16260233/185924987-f6bbda9d-ecf0-4056-b33b-5c8d835a098f.png">

Above is the installation while below is the error page
<img width="935" alt="Screenshot 2022-08-22 at 1 42 34 PM" src="https://user-images.githubusercontent.com/16260233/185925009-da14ff55-879b-4275-ae40-03b0f67963e3.png">

**Comment by kenjis** [comment]
How did you install CodeIgniter?
Did you change any Config file?

**Reply by maitalata** [reply]
I install codeigniter using composer and I did not change any config file

**Reply by kenjis** [reply]
It is weird. Cannot reproduce.

First of all, `Config\PasswordRules` does not exist.
The rule class is `CodeIgniter\Shield\Authentication\Passwords\ValidationRules`.
And it is registered by `CodeIgniter\Shield\Config\Registrar`.

**Reply by maitalata** [reply, marked as answer]
After a close look and reading the shield docs again, I am able to identify that I caused the problem myself by adding `PasswordRules::class` to the shield ruleSet array while at the same time my Codeigniter version is 4.2.4. Thank you everyone for the prompt responses.

**Reply by datamweb** [reply]
We have made some changes to the installation instructions in #401.
**Thanks.**
